Fix udp_dst and sctp_dst in flow matching string

jianjuns opened the pull request:
It impacts the OVS flow dumping implementation.

jianjuns commented:
> 
> 
> Just looked at the documentation and it seems that `tp_dst` is deprecated. You should probably revert the change for tcp as well (`MatchTCPDstPort`)?

Where you saw it is deprecated? At least the OVS version we are using print TCP port as tp_dst/tp_src?

antoninbas commented:
```
       tp_src=port
       tp_dst=port
              These are deprecated generic forms of L4 port matches.   In  new
              code,   please  use  the  TCP-,  UDP-,  or  SCTP-specific  forms
              described above.
```

http://www.openvswitch.org//support/dist-docs-2.5/ovs-ofctl.8.txt

According to this, `tp_dst` doesn't actually mean TCP dest port.

jianjuns commented:
> 
> 
> ```
>        tp_src=port
>        tp_dst=port
>               These are deprecated generic forms of L4 port matches.   In  new
>               code,   please  use  the  TCP-,  UDP-,  or  SCTP-specific  forms
>               described above.
> ```
> 
> http://www.openvswitch.org//support/dist-docs-2.5/ovs-ofctl.8.txt
> 
> According to this, `tp_dst` doesn't actually mean TCP dest port.

OK, it might mean tcp_dst should be used to add flows, but the output by ofctl is still tp_dst, so we have to use tp_dst to match the flow.

antoninbas commented:
Probably worth having a comment about this in the code though, in case this changes if we update the OVS version we are using

jianjuns commented:
> 
> 
> Probably worth having a comment about this in the code though, in case this changes if we update the OVS version we are using

Added comments. I now feel it might be a bug in OVS tool that do not change TCP port format.

jianjuns commented:
/test-all

jianjuns commented:
@antoninbas : Sorry, I mistook - even UDP and SCTP are printed as "tp_dst", so my last change is correct. I did test before, but just got confused by the different naming in ofctl/appctl input and output - the input allows tp_dst and tcp_dst for TCP, but only upd_dst/sctp_dst for UDP/SCTP; while the output always uses tp_dst for any of TCP/UDP/SCTP.
So, I reverted the change, and updated this PR to add comments only.

jianjuns commented:
/skip-all
Just added comments.
